data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/http/service/HttpSinkService.java

```diff
@@ -123,15 +134,18 @@ public HttpSinkService(final HttpSinkConfiguration httpSinkConfiguration,
         this.maxBytes = httpSinkConfiguration.getThresholdOptions().getMaximumSize();
         this.maxCollectionDuration = httpSinkConfiguration.getThresholdOptions().getEventCollectTimeOut().getSeconds();
         this.httpPluginSetting = httpPluginSetting;
-        if (httpSinkConfiguration.isSsl() || httpSinkConfiguration.useAcmCertForSSL()) {
+		this.oAuthAccessTokenManager = new OAuthAccessTokenManager();
+
+        if ((!httpSinkConfiguration.isInsecureSkipVerify()) || (httpSinkConfiguration.useAcmCertForSSL())) {
```

Comment:
@dlvenable In HttpClientConnectionManager class, we are checking ssl Certificate File exist or not, and based on that we are either using crt or Trusting All Server.

Reply:
Please elaborate some. 

When `insecure_skip_verify` is true, then the HTTP sink will not perform certificate validation against the server. If it is set to false (the default), then it will perform certification validation against the server. That is all this flag controls. There is nothing else it determines.

So when `insecure_skip_verify` is false, the next question is: How do we know whether we will trust the server? By default, Java has a trust chain. So trust is based on known root certificates. The user does not need to supply an SSL certificate for endpoints that use a certificate signed by a trusted CA. The user only needs to supply an SSL certificate when using a self-signed certificate.

Reply:
As per our understanding, if insecure_skip_verify is false, we are first checking weather ssl certificate exist or not in pipeline in createHttpClientConnectionManager() of class HttpClientSSLConnectionManager.java.
if not, we are appling TrustAllStrategy
if yes, we are reading the certificate provided in the pipeline.

Reply:
Please see my comment above. I explain what the `insecure_skip_verify` means.

The end result is that when `insecure_skip_verify` is set to `true`, the sink should not care at all about certificates. It doesn't need to load certificates at all. All you need to do when `insecure_skip_verify` is `true` is _skip the certificate validation_.

Reply:
This may be the way to do this in Apache HTTP 5. I've not verified it, but please see if it works.

https://stackoverflow.com/a/69382828/650176

Reply:
@dlvenable, changes updated in the PR